lib/search.js

```diff
@@ -4,15 +4,18 @@ module.exports = search
 
 var toString = require('mdast-util-to-string')
 var visit = require('unist-util-visit')
+var is = require('unist-util-is')
 var slugs = require('github-slugger')()
 
 var HEADING = 'heading'
 
 // Search a node for a location.
-function search(root, expression, maxDepth) {
+function search(root, expression, settings) {
   var length = root.children.length
   var depth = null
   var lookingForToc = expression !== null
+  var maxDepth = settings.maxDepth || 6
+  var parents = settings.parents || []
```

Comment:
```suggestion
  var parents = settings.parents
```

[null or undefined](https://github.com/syntax-tree/unist-util-is/blob/35b280ee14ad215280fe6da89a33275c813265ae/index.js#L41) should be a bit quicker that an empty array!